src/site/antora/modules/ROOT/pages/manual/pattern-layout.adoc

```diff
@@ -0,0 +1,1719 @@
+[source,text]
+----
+marker
+markerSimpleName
+----
+
+`marker` outputs the full name of the marker, including its parents.
+Whereas, `markerSimpleName` outputs the simple name of the marker without its parents.
+
+[#converter-map]
+==== Map
+
+Outputs the entries in a xref:manual/messages.adoc#MapMessage[`MapMessage`], if one is present in the event
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/MapPatternConverter.html[`MapPatternConverter`] specifier grammar
+[source,text]
+----
+K{key}
+map{key}
+MAP{key}
+----
+
+The `K` conversion character can be followed by the key for the map placed between braces, as in `%K\{clientNumber}`, where `clientNumber` is the key.
+The value of the map corresponding to the key will be output.
+If no additional sub-option is specified, then all map entries are output using a `{{key1,val1},{key2,val2}}` format.
+
+[#converter-max-len]
+==== Max. length
+
+Outputs the result of evaluating the given pattern and truncating the result
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/MaxLegthConverter.html[`MaxLengthConverter`] specifier grammar
+[source,text]
+----
+maxLen{pattern}{length}
+maxLength{pattern}{length}
+----
+
+If the length is greater than 20, then the output will contain a trailing ellipsis.
+If the provided length is invalid, a default value of 100 is used.
+
+For instance, `%maxLen{%p: %c\{1} - %m%notEmpty{ =>%ex\{short}}}\{160}` will be limited to 160 characters with a trailing ellipsis.
+`%maxLen{%m}\{20}` will be limited to 20 characters and no trailing ellipsis.
+
+[#converter-message]
+==== Message
+
+Outputs the message associated with the log event
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/MessagePatternConverter.html[`MessagePatternConverter`] specifier grammar
+[source,text]
+----
+m{lookups}{ansi}
+msg{lookups}{ansi}
+message{lookups}{ansi}
+----
+
+Add `\{ansi}` to render messages with ANSI escape codes.
+Windows users should refer to xref:#jansi[].
+
+The default syntax for embedded ANSI codes is:
+
+[source,text]
+----
+@\|code(,code)* text\|@
+----
+
+For example, to render the message `Hello` in green, use:
+
+[source,text]
+----
+@\|green Hello\|@
+----
+
+To render the message `Hello` in bold and red, use:
+
+[source,text]
+----
+@\|bold,red Warning!\|@
+----
+
+You can also define custom style names in the configuration with the syntax:
+
+[source,text]
+----
+%message{ansi}{StyleName=value(,value)*( StyleName=value(,value)*)*}%n
+----
+
+For example:
+
+[source,text]
+----
+%message{ansi}{WarningStyle=red,bold KeyStyle=white ValueStyle=blue}%n
+----
+
+The call site can look like this:
+
+[source,text]
+----
+logger.info("@\|KeyStyle {}\|@ = @\|ValueStyle {}\|@", entry.getKey(), entry.getValue());
+----
+
+You can enable xref:manual/lookups.adoc[] using `\{lookups}` and log messages like `${date:YYYY-MM-dd}` using lookups.
```

Comment:
Can you really evaluate _lookups_ in **messages**?
I think this functionality was removed around `2.17.x`.

Reply:
Fixed in 76f77b59edc56a27ae277dbcd8c605c0a04841ff.